### ISSUE TITLE
Reference in referenced parameter is resolved in wrong file

### ISSUE THREAD

**Issue by bug-brain** [orval-labs/orval#394]
## What are the steps to reproduce this issue?
spec.yaml
```yaml
openapi: 3.0.2
info: {}
paths:
  "/path":
    get:
      parameters:
        - $ref: "refs.yaml#/components/parameters/switch"
      responses:
        '200':
          content:
            application/json:
              schema:
                type: string
```

refs.yaml
```yaml
components:
  schemas:
    Switch:
      type: string
      enum:
        - on
        - off
  parameters:
    switch:
      name: switch
      in: query
      schema:
        $ref: "#/components/schemas/Switch"
```

## What happens?
Orval fails.

## What were you expecting to happen?
Orval completes without errors.

## Any logs, error output, etc?
```
🛑  backend - Oups... 🍻. Ref not found: #/components/schemas/Switch
```

## Any other comments?
When I added another Switch schema to spec.yaml code was generated but the parameter from refs.yaml was used.

## What versions are you using?

**Package Version:** 6.7.1

**Comment by abettermap** [comment]
Same issue here.

## So far I have tried:

* Pointing the `$ref` to a a relative path in the project root using `./spec.json`
* Pointing the `$ref` to a a relative path in the project root using `spec.json` (no leading period)
* Pointing the `$ref` to a relative path in a nested directory
* Pointing the `$ref` to an absolute path, e.g. `/Users/me/something.../spec.json`
* Pointing the `$ref` to an external web URL, e.g. `https://something.../spec.yaml`
* Using a spec in `yaml` format
* Using a spec in `JSON` format
* Using a `$ref` with a `$1` instead of `/` for the nested ref, e.g. `spec.yaml#/components$1/schemas/Geometry`
* Using a `$ref` with `/` characters intact
* Pointing to a `$ref` in the same folder as my `transformer`
* Pointing to a `$ref` in a different folder

If it helps, here is my **config**:

```typescript
// eslint-disable-next-line import/no-extraneous-dependencies
import { defineConfig } from 'orval'

import transformer from './src/lib/api/django/overrides.django'

export default defineConfig({
  'django-backend': {
    input: {
      target: './scratch/portal-api.json',
      converterOptions: true,
      override: { transformer },
    },
    output: {
      client: 'react-query',
      mock: false, // TODO
      mode: 'tags-split',
      schemas: 'src/lib/api/django/model',
      target: 'src/lib/api/django/types.django-api.ts',
      prettier: true,
      override: {
        mutator: {
          path: 'src/lib/api/django-axios-instance.ts',
          name: 'djangoAxiosInstance',
        },
      },
    },
  },
})
```

And my **transformer**:

```typescript
// eslint-disable-next-line import/no-extraneous-dependencies
import { OpenAPIObject, SchemaObject } from 'openapi3-ts'

function transformer(inputSchema: OpenAPIObject): OpenAPIObject {
  const { components } = inputSchema
  const { schemas } = components || {}
  const { Field } = schemas || ({ properties: {} } as SchemaObject)

  return {
    ...inputSchema,
    components: {
      ...inputSchema.components,
      schemas: {
        ...schemas,
        Field: {
          ...Field,
          properties: {
            ...Field.properties,
            geometry: {
              oneOf: [
                // None of these work 😞
                { $ref: 'geojson-spec.yaml#/components~1schemas~1Point' },
                { $ref: 'geojson-spec.yaml#/components~1schemas~1LineString' },
                { $ref: 'geojson-spec.yaml#/components~1schemas~1Polygon' },
                { $ref: 'geojson-spec.yaml#/components~1schemas~1MultiPoint' },
                { $ref: 'geojson-spec.yaml#/components~1schemas~1MultiLineString' },
                { $ref: 'geojson-spec.yaml#/components~1schemas~1MultiPolygon' },
              ],
            },
          },
        },
      },
    },
  }
}

export default transformer
```

## It works if I hard-code the refs:

...into the other file (`'./scratch/portal-api.json'` in my config above) using the `oneOf` syntax.

Hardcoding isn't a realistic option, of course, I just wanted to point out that `orval` works except for using refs in the `transformer` (other overrides worked fine).

I don't know enough about the source code for a PR, but I love this repo and appreciate the work so I'd be happy to troubleshoot/answer questions! Maybe it will alleviate some of the other issues that pop up in [a "ref not found" search](https://github.com/anymaniax/orval/search?q=ref+not+found&type=issues).

**Comment by HugoPoi** [comment]
As workaround for this you might add `redocly bundle spec.yaml --ext json -o spec.json` and input is the spec.json file.
Orval doesn't have a hook like `beforeReadSpecs` but you could add a scripts command like `"generate-api-clients": "redocly bundle spec.yaml --ext json -o spec.json && orval --config orval.config.ts"`

**Comment by wadakatu** [comment]
Quick triage on the latest version (v8.9.1).

I reproduced the original report from @bug-brain and it now generates correctly:

**spec.yaml**
```yaml
openapi: 3.0.2
info: { title: repro, version: '1.0.0' }
paths:
  /path:
    get:
      operationId: getPath
      parameters:
        - $ref: "refs.yaml#/components/parameters/switch"
      responses:
        '200':
          description: ok
          content:
            application/json:
              schema: { type: string }
```

**refs.yaml**
```yaml
components:
  schemas:
    Switch:
      type: string
      enum: [on, off]
  parameters:
    switch:
      name: switch
      in: query
      schema:
        $ref: "#/components/schemas/Switch"
```

**Generated output (excerpt):**
```ts
export const Switch = { on: 'on', off: 'off' } as const;
export type GetPathParams = { switch?: Switch };
```

I also confirmed these related cases work in v8.9.1:
- Name collision (same `Switch` schema in both files) — auto-suffixed to avoid clash
- External `$ref` for `responses` and `requestBodies`
- Three-level chain (`spec.yaml` → `refs.yaml` → `deep.yaml`)

This appears to have been resolved by #2944, which introduced external-spec bundling/dereferencing via `@scalar/json-magic`. There was no dedicated regression test for this specific shape (external-file `parameters` whose inner `$ref` points to a schema in the same external file), so I opened #3324 to lock it in.

---

Regarding @abettermap's comment: I believe that's actually a **different problem**, even though the symptom looks similar. There the `$ref`s are injected at runtime by `override.transformer`, which runs **after** the bundling step — so the new external refs are never bundled and end up unresolved (`Can't resolve external reference: refs.yaml` on current `master`). I'd suggest filing it as a separate issue so it can be tracked independently.

@melloware I'd appreciate your take here — if you agree, this issue could probably be closed once #3324 is merged. Happy to open a follow-up issue for the transformer case as well.

**Comment by melloware** [comment]
I totally agree let me link your test case and close this ticket and if there is a different separate more specific issue it should be opened.
